docs/upgrading.md

```diff
@@ -0,0 +1,111 @@
+# Upgrading Clear Containers installations
+
+This document discusses the available options for upgrading a system running Intel® Clear Containers.
+
+## Terminology
+
+There are currently two implementations of Clear Containers:
+
+- Clear Containers 2.x (or just 2.x) refers to the previous implementation
+  of Clear Containers which uses a runtime called `cc-oci-runtime` and
+  whose code is available at: https://github.com/01org/cc-oci-runtime
+
+- Clear Container 3.x (or just 3.x) refers to the current implementation
+  which uses a runtime called `cc-runtime` and whose code is available
+  at: https://github.com/clearcontainers/runtime
+
+## Maintenance warning
+
+The 2.x codebase is no longer being developed. New releases will *only* be
+considered for significant bugs.
+
+The main development focus is now on the 3.x branch. As such, all 2.x
+users are encouraged to switch to the 3.x implementation.
+
+## Introduction
+
+The two Clear Containers implementations share certain architectural
+elements. Namely, both use a shim process called `cc-shim` and a proxy
+process called `cc-proxy`. However, the 2.x versions of these
+elements *are incompatible* with their 3.x counterparts.
+
+## Supported operations and configurations
+
```

Comment:
Include a description under a section. We want to avoid having two headings one after another.